shared/middleware/files_api.rb

```diff
@@ -477,9 +477,15 @@ def files_put_file(encrypted_channel_id, filename, body)
       manifest = JSON.load manifest_result[:body]
     end
 
-    # store the new file
+    # store the new file (possibly copied from an existing file)
     if params['src']
-      new_entry_json = copy_file('files', encrypted_channel_id, filename.downcase, params['src'])
+      if filename.downcase == params['src']
+        # The rename is a case-only rename. As the filename is downcased on s3, there is no need to
+        # update s3.
+        new_entry_json = manifest.detect {|e| e['filename'].downcase == filename.downcase}.to_json
```

Comment:
Can we move the `filename.downcase` call out of the loop - maybe even do this once outside the `if` statement and reuse the result?